x-pack/plugins/stack_connectors/server/connector_types/openai/lib/azure_openai_utils.ts

```diff
@@ -61,8 +61,6 @@ export const getRequestWithStreamOption = (url: string, body: string, stream: bo
 export const transformApiUrlToRegex = (apiUrl: string): RegExp => {
   return new RegExp(
     apiUrl
-      .replaceAll(`/`, `\/`)
-      .replaceAll(`.`, `\.`)
```

Comment:
__question:__ I don't know the intention of this function, but shouldn't we change this to something like this instead?
```ts
      .replaceAll(`/`, `\\/`)
      .replaceAll(`.`, `\\.`)
```

Otherwise, please correct me if I'm wrong, we'll match URLs even if instead of `.` they use something else. For example, the resulting regular expression will match both.
`https://My-test-resource-123.openai.azure.com/openai...` (valid URL) and `https://My-test-resource-123.openaiaazure.com/openai...` (not valid, potentially malicious domain name, notice `openaiaazure.com`)


The fact that tests pass with this change makes me wonder if the tests are correct :) Do we test **un**happy path too?

Reply:
Thanks for the review! Updated and added more unhappy path tests.